File: 153-find-minimum-in-rotated-sorted-array/find-minimum-in-rotated-sorted-array.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        int min_element = nums[0];
        int i=0, j=nums.size()-1;
        while(i <= j){
            int mid = midpoint(i,j);

            if (nums[i] <= nums[j]){
                min_element = min(nums[i], min_element);
                break;
            }

            min_element = min(min_element, nums[mid]);

            if (nums[i] <= nums[mid]){
                i = mid + 1;
            } else {
                j = mid - 1;
            }
        }
        return min_element;
    }
};

File: 153-find-minimum-in-rotated-sorted-array/find-minimum-in-rotated-sorted-array.go
func findMin(nums []int) int {
    min_element := nums[0]
    left, right := 0, len(nums)-1

    for left <= right {
        mid := left + (right - left)/2

        if nums[left] <= nums[right] {
            min_element = min(min_element, nums[left])
            break
        }

        min_element = min(min_element, nums[mid])

        if nums[left] <= nums[mid] {
            left = mid + 1
        } else {
            right = mid - 1
        }
    }

    return min_element
}